rust/operator-binary/src/restart_controller.rs

```diff
@@ -0,0 +1,308 @@
+use std::collections::BTreeMap;
+use std::sync::atomic::AtomicBool;
+use std::sync::Arc;
+use std::time::Duration;
+
+use futures::{stream, Stream, StreamExt, TryStream};
+use serde_json::json;
+use snafu::Snafu;
+use stackable_operator::client::Client;
+use stackable_operator::k8s_openapi::api::apps::v1::StatefulSet;
+use stackable_operator::k8s_openapi::api::core::v1::{
+    ConfigMap, EnvFromSource, EnvVar, PodSpec, Secret, Volume,
+};
+use stackable_operator::kube;
+use stackable_operator::kube::api::{ListParams, Patch, PatchParams};
+use stackable_operator::kube::core::DynamicObject;
+use stackable_operator::kube::runtime::controller::{
+    trigger_self, trigger_with, Action, Context, ReconcileRequest,
+};
+use stackable_operator::kube::runtime::reflector::{ObjectRef, Store};
+use stackable_operator::kube::runtime::utils::{try_flatten_applied, try_flatten_touched};
+use stackable_operator::kube::runtime::{applier, reflector, watcher};
+use stackable_operator::kube::{Resource, ResourceExt};
+use stackable_operator::logging::controller::{report_controller_reconciled, ReconcilerError};
+use strum::{EnumDiscriminants, IntoStaticStr};
+
+struct Ctx {
+    kube: kube::Client,
+    cms: Store<ConfigMap>,
+    cms_inited: Arc<AtomicBool>,
+    secrets: Store<Secret>,
+    secrets_inited: Arc<AtomicBool>,
+}
+
+#[derive(Snafu, Debug, EnumDiscriminants)]
+#[strum_discriminants(derive(IntoStaticStr))]
+enum Error {
+    #[snafu(display("tried to reconcile {} which has no namespace", obj_ref))]
+    ObjectHasNoNamespace { obj_ref: ObjectRef<DynamicObject> },
+    #[snafu(display("failed to apply update to {}", obj_ref))]
+    Apply {
+        source: kube::Error,
+        obj_ref: ObjectRef<DynamicObject>,
+    },
+    #[snafu(display("configmaps were not yet loaded"))]
+    ConfigMapsUninited,
+    #[snafu(display("secrets were not yet loaded"))]
+    SecretsUninited,
+}
+
+impl ReconcilerError for Error {
+    fn category(&self) -> &'static str {
+        ErrorDiscriminants::from(self).into()
+    }
+}
```

Comment:
```suggestion
impl ReconcilerError for Error {
    fn category(&self) -> &'static str {
        ErrorDiscriminants::from(self).into()
    }

    fn secondary_object(&self) -> Option<ObjectRef<DynamicObject>> {
        match self {
            Error::ObjectHasNoNamespace { obj_ref, .. } => {
                Some(obj_ref.clone().erase())
            }
            Error::MissingDruidConnString => None,
            Error::Apply { obj_ref, .. } => Some(obj_ref.clone().erase()),
            Error::ConfigMapsUninited { .. } => None,
            Error::SecretsUninited { .. } => None,
        }
    }
}
```

Reply:
Thanks for the hint! I implemented a little bit different, hope that's ok